source/common/singleton/manager_impl.cc

```diff
@@ -9,7 +9,7 @@ namespace Envoy {
 namespace Singleton {
 
 InstanceSharedPtr ManagerImpl::get(const std::string& name, SingletonFactoryCb cb) {
-  ASSERT(run_tid_->isCurrentThreadId());
+  // ASSERT(run_tid_->isCurrentThreadId());
```

Comment:
any particular reason not to provide this API?

Reply:
yes, the ThreadFactory can answer it for you, but the Id cannot

Reply:
We need some sort of alternative solution then rather than just commenting it out.

Reply:
I have two thoughts on this.

1) it is as an assert, so technically we could just drop it (I don't really like this option but would do it if needed.
2) if we have access to the ThreadFactory here we can do it pretty easily, but I am not sure how to get it here

I don't want to add a pointer to the factory to the ID just to enable this assert

Reply:
I think this assert is considered important as envoy's silo'd threading model does depend on this.

Moreover, why do we need to have the manager involved implementing this? The current code is just
```
bool ThreadIdImplPosix::isCurrentThreadId() const { return id_ == getCurrentThreadId(); }
```
which seems OK in the current architecture.

Reply:
this design does not have a ThreadIdImplPosix class at all

Reply:
One way or the other IMO I think we should be able to do the assert in this location as part of this PR. Options include:
 - add the factory to the thread-id object
 - make the factory method to get the current thread id be static so you can call it from here
 - plumb the factory into the places where we do the assert.

Reply:
1) I dislike as wasteful, but will yield if I cannot accomplish this another way
2) static + virtual do not interact nicely, so I think it is a non-starter
3) this is the only place we do the assert that did not already have the factory at hand.  I would be happy to plumb it to here, but I don't know the envoy code base that well.  Pointers would be greatly appreciated.

Reply:
up to you. option 3 sounds fine; if it's only one place, I'd say wherever we are populating the thread ID being held you could also populate the factory there; sounds reasonably surgical.

Reply:
Done.